stellarphot/settings/tests/test_tess_photometry_setup.py

```diff
@@ -0,0 +1,136 @@
+        widget.drop.value = 0
+
+        # The TIC ID entry should be visible
+        assert widget.tic_id_entry.layout.display == "flex"
+        # The FITS opener should be hidden
+        assert widget.fits_opener.file_chooser.layout.display == "none"
+
+        # now choose the FITS opener method
+        widget.drop.value = 1
+
+        # The TIC ID entry should be hidden
+        assert widget.tic_id_entry.layout.display == "none"
+        # The FITS opener should be visible
+        assert widget.fits_opener.file_chooser.layout.display == "flex"
+
+    def test_set_tic_id_via_text_box(self, tess_tic_expected_values):
+        widget = TessPhotometrySetup()
+        # This needs to be a real TIC ID....
+        widget.tic_id_entry.value = tess_tic_expected_values["tic_id"]
+        assert widget.tic_id == tess_tic_expected_values["tic_id"]
+
+        # The confirmation dialog should be visible now
+        assert widget.confirm.layout.display == "flex"
+
+    def test_get_tic_id_via_file(self, tess_tic_expected_values):
+        widget = TessPhotometrySetup()
+
+        # Make a little FITS file with an object keyword
+        hdu = fits.PrimaryHDU()
+        hdu.header["object"] = f"TIC {tess_tic_expected_values['tic_id']}"
+
+        fits_file = Path("test.fits")
+
+        hdu.writeto(fits_file)
+
+        widget.fits_opener.file_chooser.value = str(fits_file)
+
+        assert widget.tic_id == tess_tic_expected_values["tic_id"]
+
+        # The confirmation dialog should be visible now
+        assert widget.confirm.layout.display == "flex"
+
+    def test_click_no_confirm(self, tess_tic_expected_values):
+        # Test that clicking "No" on the confirmation dialog hides the dialog
+        # and does not display the "All Done" display
+        widget = TessPhotometrySetup()
+        widget.tic_id_entry.value = tess_tic_expected_values["tic_id"]
+
+        # Confirmation box is displayed, click no
+        widget.confirm._no.click()
+
+        # The confirmation dialog should be hidden now
+        assert widget.confirm.layout.display == "none"
+
+        # The all done display should still be hidden
+        assert widget.all_done.layout.display == "none"
+
+    @pytest.mark.remote_data
+    def test_click_yes_confirm(self, tess_tic_expected_values):
+        # Test that clicking "Yes" on the confirmation dialog hides the dialog
+        # and displays the "All Done" display and creates the files we expect.
+
+        widget = TessPhotometrySetup()
+        widget.tic_id_entry.value = tess_tic_expected_values["tic_id"]
+
+        # Confirmation box is displayed, click yes
+        widget.confirm._yes.click()
+
+        # The confirmation dialog should be hidden now
+        assert widget.confirm.layout.display == "none"
+
+        # The all done display should be visible
+        assert widget.all_done.layout.display == "flex"
+
+        # There should be exactly one json file from which we can create a TOI object
```

Comment:
For testing this is fine, since you are in a temp directory, but we can't assume this normally, right?

Reply:
Correct -- I just didn't want to hard code the file name.